services/task-svc/internal/room/room.go

```diff
@@ -23,6 +23,24 @@ func NewServiceServer() *ServiceServer {
 	return &ServiceServer{}
 }
 
+// GetRoomsWithBedsAndPatientsAndTasksByWardForOrganization
+// TODO: Move into repository
+func GetRoomsWithBedsAndPatientsAndTasksByWardForOrganization(ctx context.Context, wardID uuid.UUID) ([]models.Room, error) {
```

Comment:
This function should never be near of a repository. I guess to have some "heavier" queries inside the "controller" is fine.